vendor/forms.py

```diff
@@ -387,6 +391,62 @@ def clean(self):
 
         return cleaned_data
 
+
+class SiteSelectForm(forms.Form):
+    site = forms.ModelChoiceField(queryset=Site.objects.all())
+
+class OfferSiteSelectForm(SiteSelectForm):
+    offer = forms.ModelChoiceField(queryset=None, required=False)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if 'site' in self.initial or 'site' in self.data:
+            site = self.initial.get('site') if self.initial.get('site', None) else self.data['site']
```

Comment:
this could be simplified a little with `site = self.initial.get('site', self.data['site'])`